src/arm.c

```diff
@@ -283,6 +285,16 @@ int __sb(arm_cond_t cond, arm_reg rd, arm_reg rn, int ofs)
     return arm_transfer(cond, 0, 1, rn, rd, ofs);
 }
 
+int __push(arm_cond_t cond, int reg_list)
```

Comment:
Both `push` and `pop` are pseudo-instructions. Can you avoid them?

Reply:
After reading [ARM Architecture Reference Manual](https://developer.arm.com/documentation/ddi0406/latest/), I understand that `push` and `pop` are equivalent to `stmdb` and `ldmia`, respectively. But I don't see any description stating that `push` and `pop` are pseudo-instructions.

May I ask where it describes `push` and `pop` as pseudo-instructions? Although I can replace them with `stmdb` and `ldmia`, I want to understand `push` and `pop` further first before improving the implementation.

Reply:
> May I ask where it describes `push` and `pop` as pseudo-instructions? 

See https://developer.arm.com/documentation/dui0489/i/arm-and-thumb-instructions/pop

Both `__push` and `__pop` are too generic, likely leading to unexpected symbol conflicts.

Reply:
>Both push and pop are pseudo-instructions. Can you avoid them?

I think I misunderstood at the beginning. Originally,  I thought it was related to code generation for Arm32 output.

However, the requested changes just entail that I should replace `__push` and `__pop` with `__stmdb` and `__ldm`  in `arm.c` and `arm-codegen.c`. The purpose is to avoid symbol conflicts during compilation.

Is my understanding correct?

Reply:
> However, the requested changes just entail that I should replace `__push` and `__pop` with `__stmdb` and `__ldm` in `arm.c` and `arm-codegen.c`. The purpose is to avoid symbol conflicts during compilation.

Exactly. Unless we can change symbol visibility on demand, we must handle symbol conflicts with care.